ee/clickhouse/models/property.py

```diff
@@ -225,50 +227,60 @@ def get_property_values_for_key(key: str, team: Team, value: Optional[str] = Non
     )
 
 
-def filter_element(filters: Dict, prepend: str = "") -> Tuple[str, Dict]:
+def filter_element(filters: Dict, *, operator: Optional[OperatorType] = None, prepend: str = "") -> Tuple[str, Dict]:
+    if not operator:
+        operator = "exact"
+
     params = {}
     conditions = []
 
     if filters.get("selector"):
-        or_conditions = []
         selectors = filters["selector"] if isinstance(filters["selector"], list) else [filters["selector"]]
         for idx, query in enumerate(selectors):
             selector = Selector(query, escape_slashes=False)
             key = f"{prepend}_{idx}_selector_regex"
             params[key] = _create_regex(selector)
-            or_conditions.append(f"match(elements_chain, %({key})s)")
-        if len(or_conditions) > 0:
-            conditions.append("(" + (" OR ".join(or_conditions)) + ")")
+            conditions.append(f"match(elements_chain, %({key})s)")
+        if len(conditions) > 0:
+            conditions.append("(" + (" OR ".join(conditions)) + ")")
 
     if filters.get("tag_name"):
-        or_conditions = []
         tag_names = filters["tag_name"] if isinstance(filters["tag_name"], list) else [filters["tag_name"]]
         for idx, tag_name in enumerate(tag_names):
             key = f"{prepend}_{idx}_tag_name_regex"
             params[key] = rf"(^|;){tag_name}(\.|$|;|:)"
-            or_conditions.append(f"match(elements_chain, %({key})s)")
-        if len(or_conditions) > 0:
-            conditions.append("(" + (" OR ".join(or_conditions)) + ")")
+            conditions.append(f"match(elements_chain, %({key})s)")
+        if len(conditions) > 0:
+            conditions.append("(" + (" OR ".join(conditions)) + ")")
 
     attributes: Dict[str, List] = {}
-
     for key in ["href", "text"]:
-        vals = filters.get(key)
-        if filters.get(key):
-            attributes[key] = [re.escape(vals)] if isinstance(vals, str) else [re.escape(text) for text in filters[key]]
-
-    if len(attributes.keys()) > 0:
-        or_conditions = []
+        if vals := filters.get(key):
```

Comment:
Code style nit: IMO relying in None here is shaky and makes this hard to follow since vals is only used N lines down.

How about:

```
if key in filters
```

However looking at the code a bit more I think it could benefit from extracting a function.

```
if key in filters:
  attributes[key] = get_element_match(filters[key], operator or "exact")

def get_element_match(values: Union[str, List[str]], operator: OperatorType):
  # Do the if-else here
```